browser-test/src/applicant_application_index.test.ts

```diff
@@ -90,4 +94,56 @@ describe('applicant program index page', () => {
       wantSubmittedPrograms: [],
     })
   })
+
+  it('shows previously answered on text for program that was previously applied to', async () => {
+    const {page, applicantQuestions} = ctx
+
+    await loginAsTestUser(page)
+    await selectApplicantLanguage(
+      page,
+      'English',
+      /* assertProgramIndexPage= */ true,
+    )
+
+    // Fill out application with one question and confirm it was successfully submitted.
+    await applicantQuestions.applyProgram(otherProgramName)
+    await applicantQuestions.answerTextQuestion('first answer')
+    await applicantQuestions.clickNext()
+    await applicantQuestions.submitFromReviewPage()
+    await applicantQuestions.returnToProgramsFromSubmissionPage()
+    await applicantQuestions.expectPrograms({
+      wantNotStartedPrograms: [primaryProgramName],
+      wantInProgressPrograms: [],
+      wantSubmittedPrograms: [otherProgramName],
+    })
+
+    // Check that the question repeated in the program with two questions shows previously answered.
+    await applicantQuestions.clickApplyProgramButton(primaryProgramName)
+    await validateScreenshot(page, 'question-shows-previously-answered')
+
+    // Fill out second question.
+    await applicantQuestions.clickContinue()
+    await applicantQuestions.answerTextQuestion('second answer')
+    await applicantQuestions.clickNext()
+    await applicantQuestions.submitFromReviewPage()
+
+    // Check that the original program doesn't show previously answered.
+    await applicantQuestions.returnToProgramsFromSubmissionPage()
+    await applicantQuestions.clickApplyProgramButton(otherProgramName)
+    await validateScreenshot(page, 'question-does-not-show-previously-answered')
```

Comment:
**issue (blocking):** We should only rely on screenshots for rendering validation and not correctness.  Changes the visual styling can easily be noise to hide correctness bugs.

For correctness we inspect the dom to ensure that it is constructed as expected.  

You'll see examples in other tests where we question the dom and assert it's state.

Reply:
Makes sense! Updated to add a dom assertion instead.